lib/rspec/mocks/message_expectation.rb

```diff
@@ -556,6 +561,7 @@ def invoke_incrementing_actual_calls_by(increment, allowed_to_fail, parent_stub,
           args.unshift(orig_object) if yield_receiver_to_implementation_block?
 
           if negative? || (allowed_to_fail && (@exactly || @at_most) && (@actual_received_count == @expected_received_count))
+            @raised = true
```

Comment:
I think I prefer `@failed` for the name, as we've note yet raised, and there are circumstances where we don't until later.